extension.js

```diff
@@ -321,12 +325,39 @@ function workarea_for_window(win) {
 }
 
 function target_rect_for_workarea(workarea) {
-    const target_rect = workarea.copy();
-    target_rect.height *= settings.get_double('window-height');
-    return target_rect;
+    const full_width = workarea.width + workarea.x,
+        full_height = workarea.height + workarea.y,
```

Comment:
Again, `workarea.height + workarea.y` isn't a "full height" of anything. Sometimes it matches the height of the monitor, but only if there is no bottom panel.

Please rename this variable, or remove it - it's only used once.

Exactly the same for `full_width`

Reply:
Done.

I'm not much familiar with gjs so a stupid question: How can I actually get the full height and width of the monitor?

Reply:
https://gjs-docs.gnome.org/meta7~7_api/meta.display#method-get_monitor_geometry

However, you shouldn't need it